src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -416,28 +415,135 @@ def _check_private_endpoint(cmd, registry_name, vnet_of_private_endpoint):  # py
+            username=username,
+            password=password,
+            json_payload=None,
+            file_payload=None,
+            manifest_headers={'Accept': 'application/vnd.docker.distribution.manifest.v2+json'},
+            params=image)
+    except CLIError as e:
+        # Error is due to the image tag not being found on manifest
+        if "manifest tagged by" in str(e):
+            raise ResourceNotFoundError(IMAGE_INPUT_ERROR_MESSAGE)
+        # Error is due to the repository not being found
+        else:
+            raise ResourceNotFoundError(REPOSITORY_INPUT_ERROR_MESSAGE)
+
+# Validate if specific blob can be pulled
+def _check_blob_exists(cmd,
+                       registry_name,
+                       repository_name,
+                       image):
+    
+    import requests
+    from .repository import get_access_credentials
+
+    registry, _ = get_registry_by_name(cmd.cli_ctx, registry_name)
+    login_server = registry.login_server.rstrip('/')
+
+    # Get the access credentials for the registry
+    registry_name, username, password = get_access_credentials(
+        cmd,
+        registry_name=registry_name,
+        tenant_suffix=None,
+        username=None,
+        password=None,
+        repository=repository_name,
+        permission='pull')
+    
+    # Get manifest
+    # GET {url}/v2/{name}/manifests/{reference}
+    manifest_url = 'https://{}/v2/{}/manifests/{}'.format(login_server, repository_name, image)
+    
+    response = requests.get(
+        manifest_url,
+        headers={'Accept': 'application/vnd.oci.artifact.manifest.v1+json'
+            ', application/vnd.cncf.oras.artifact.manifest.v1+json'
+            ', application/vnd.oci.image.manifest.v1+json'
+            ', application/vnd.oci.image.index.v1+json'
+            ', application/vnd.docker.distribution.manifest.v2+json'
+            ', application/vnd.docker.distribution.manifest.list.v2+json'},
+        auth=(username, password))
+    
+    manifest = response.json()
+
+    # Get the digest of the smallest blob for performance purposes
+    def get_smallest_blob_digest(manifest):
+       
+        smallest_blob = min(manifest['layers'], key=lambda layer: layer['size'])
+        smallest_blob_digest = smallest_blob['digest']
+
+        return smallest_blob_digest
+
+    digest = get_smallest_blob_digest(manifest)
+
+    # Pull blob - GET {url}/v2/{name}/blobs/{digest}
+    request_url = 'https://' + login_server + '/v2/' + repository_name + '/blobs/' + digest
+    
+    logger.debug(add_timestamp("Sending a HTTP GET request to {}".format(request_url)))
+    
+    response = requests.get(request_url, auth=(username, password))
+    
+    if response.status_code < 400:
+        print_pass("Blob identified by image '{}' is available to pull from registry '{}'".format(image, login_server))
+    else:
+        raise CLIError("'Blob identified by image '{}' cannot be pulled from registry '{}'".format(image, login_server)) 
```

Comment:
Do we want to give the user the response status code or the error text returned from backend?

Reply:
If status code is 400 and above, the returned message will come from backend.